src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,775 @@
+
+    double trace = arma::accu(arma::pow(rgInitialStds, 2));
+    sigma = std::sqrt(trace/N);
+
+    chiN = std::sqrt((double) N) * (1 - 1/(4*N) + 1/(21*N*N));
+    eigensysIsUptodate = true;
+    doCheckEigen = false;
+    genOfEigensysUpdate = 0;
+
+    double dtest;
+    for (dtest = double(1); dtest && dtest < double(1.1)*dtest; dtest *= double(2))
+      if (dtest == dtest + double(1))
+        break;
+    dMaxSignifKond = dtest / double(1000);
+
+    gen = 0;
+    countevals = 0;
+    state = INITIALIZED;
+    dLastMinEWgroesserNull = double(1);
+
+    pc.set_size(N);
+    ps.set_size(N);
+    tempRandom.set_size(N+1);
+    BDz.set_size(N);
+    xmean.set_size(N+2);
+    xold.set_size(N+2);
+    xold[0] = N;
+    ++xold;
+    xBestEver.set_size(N+3);
+    xBestEver[0] = N;
+    ++xBestEver;
+    xBestEver[N] = std::numeric_limits<double>::max();
+    rgD.set_size(N);
+    C.set_size(N,N);
+    B.set_size(N,N);
+    publicFitness.set_size(lambda);
+    functionValues.set_size(lambda+1);
+    functionValues[0] = lambda;
+    ++functionValues;
+    const int historySize = 10 + (int) ceil(3.*10.*N/lambda);
+    funcValueHistory.set_size(historySize + 1);
+    index = arma::linspace<arma::uvec>(0, lambda-1, lambda);
+    population.zeros(lambda, N+2);
+    functionValues.fill(DBL_MAX);
+    funcValueHistory.fill(DBL_MAX);
+    funcValueHistory[0] = (double) historySize;
+    funcValueHistory++;
+  
+    C.zeros();
+    B.zeros();
+    B.diag().ones();
+
+    rgD = rgInitialStds * std::sqrt(N / trace);
+    C.diag() = rgD;
+    arma::pow(C.diag(),2);
+    pc.zeros();
+    ps.zeros();
+
+    minEW = rgD.min();
+    minEW = minEW*minEW;
+    maxEW = rgD.max();
+    maxEW = maxEW*maxEW;
+
+    maxdiagC = arma::max(C.diag());
+    mindiagC = arma::min(C.diag());
+  
+      xmean = xold;
+      xmean = xstart;
+    
+    if (typicalXcase)
+     xmean += sigma * rgD * rand.gauss();
+
+    func.subvec(0, lambda - 1) = publicFitness.subvec(0, lambda - 1);
+  }
+
+  /**
+   * The search space vectors have a special form: they are arrays with N+1
+   * entries. Entry number -1 is the dimension of the search space N.
+   * @return A pointer to a "population" of lambda N-dimensional multivariate
+   * normally distributed samples.
+   */
+   template<typename funcType>
+   void CMAES<funcType>::samplePopulation()
+  {
+    bool diag = diagonalCov == 1 || diagonalCov >= gen;
+
+    // calculate eigensystem
+    if (!eigensysIsUptodate)
+    {
+      if (!diag)
+        updateEigensystem(false);
+      else
+      {
+        for (int i = 0; i < N; ++i)
+          rgD[i] = std::sqrt(C(i,i));
+        minEW = rgD.min();
+        minEW *= minEW;
+        maxEW = rgD.max();
+        maxEW *= maxEW;
+        eigensysIsUptodate = true;
+      }
+    }
+
+    for (int iNk = 0; iNk < lambda; ++iNk)
+    { // generate scaled random vector D*z
+      for (int i = 0; i < N; ++i)
+        if (diag)
+          population(iNk,i) = xmean[i] + sigma*rgD[i]*rand.gauss();
```

Comment:
It's true, that if you rerun the executable or test you get the same random numbers, you can set ``math::RandomSeed(std::time(NULL));`` at the beginning if you like to see different numbers without rebuilding. But I think for now it's unnecessary since calling e.g. ``randn`` multiple times returns different values:

```
for (size_t i = 0; i < 5; ++i)
    std::cout << arma::randu(10, 1) << std::endl;
```
What do you think?